src/auto-evo/RunResults.cs

```diff
@@ -811,6 +826,144 @@ void OutputPopulationForPatch(Species species, Patch patch, long population)
             return builder;
         }
 
+        public void LogResultsToTimeline(GameWorld world, List<ExternalEffect> effects = null)
+        {
+            var newSpecies = GetNewSpecies();
+
+            foreach (var patch in world.Map.Patches.Values)
+            {
+                foreach (var species in patch.SpeciesInPatch.Keys)
+                {
+                    var hasPopulationBeforeMigration = GetSpeciesPopulationsByPatch(species, true, true)
+                        .TryGetValue(patch, out var finalPatchPopulation);
+
+                    var globalPopulation = GetGlobalPopulation(species, true, true);
+                    var previousGlobalPopulation = world.Map.GetSpeciesGlobalPopulation(species);
+
+                    var previousPatchPopulation = patch.GetSpeciesPopulation(species);
+
+                    // Adjust final patch population by external effects
+                    if (effects != null)
+                    {
+                        foreach (var effect in effects)
+                        {
+                            if (effect.Species == species)
+                            {
+                                finalPatchPopulation += effect.Constant + (long)(effect.Species.Population *
+                                    effect.Coefficient) - effect.Species.Population;
+                            }
+                        }
+                    }
+
+                    if (globalPopulation <= 0)
+                    {
+                        // TODO: see https://github.com/Revolutionary-Games/Thrive/issues/2958
+                        LogEventGloballyAndLocally(world, patch, new LocalizedString(
+                                "TIMELINE_SPECIES_EXTINCT", species.FormattedName),
+                            species.PlayerSpecies, "extinction.png");
```

Comment:
using a continue here would reduce nesting for quite a bit of code that's in the else.